Source/Orts.Simulation/Simulation/RollingStocks/TrainCar.cs

```diff
@@ -3236,17 +3238,58 @@ public virtual float GetBrakeShoeFrictionFactor()
                 // https://iopscience.iop.org/article/10.1088/1742-6596/1614/1/012086/pdf
+                //
+                // These formulas are based upon the work of Karwatzki, which is described further in "Fahrdynamik des Schienenverkehrs" by Dietrich Wende 2003.
+                // Karwatzki developed a formula for brakeshoe friction that used 5 constant values k1, ..... , k5.
+                // Wende present a number of different brake shoes k values, two sets of which line up with the UIC values above.
 
                 float NewtonsTokNewtons = 0.001f;
                 float brakeShoeForcekN = (NewtonsTokNewtons * BrakeShoeForceN) / NumberCarBrakeShoes;
+                float k1 = 0;
+                float k2 = 0;
+                float k3 = 0;
+                float k4 = 0;
+                float k5 = 0;
 
                 if (BrakeShoeType == BrakeShoeTypes.Cast_Iron)
                 {
-                    frictionfraction = 0.6f * ((1.6f * brakeShoeForcekN + 100.0f) / (8.0f * brakeShoeForcekN + 100.0f)) * ((MpS.ToKpH(AbsSpeedMpS) + 100.0f) / (5.0f * MpS.ToKpH(AbsSpeedMpS) + 100.0f));
+                    k1 = 0.024f;
+                    k2 = 62.5f;
+                    k3 = 12.5f;
+                    k4 = 100f;
+                    k5 = 20f;
+
+                    frictionfraction = k1 * ((brakeShoeForcekN + k2) / (brakeShoeForcekN + k3)) * ((MpS.ToKpH(AbsSpeedMpS) + k4) / (MpS.ToKpH(AbsSpeedMpS) + k5));
+                }
+                else if (BrakeShoeType == BrakeShoeTypes.Cast_Iron_P10)
+                {
+                    k1 = 0.05f;
+                    k2 = 62.5f;
+                    k3 = 31.25f;
+                    k4 = 100f;
+                    k5 = 20f;
+
+                    frictionfraction = k1 * ((brakeShoeForcekN + k2) / (brakeShoeForcekN + k3)) * ((MpS.ToKpH(AbsSpeedMpS) + k4) / (MpS.ToKpH(AbsSpeedMpS) + k5));
                 }
                 else if (BrakeShoeType == BrakeShoeTypes.High_Friction_Composite)
                 {
-                    frictionfraction = 0.44f * ((0.1f * brakeShoeForcekN + 20.0f) / (0.4f * brakeShoeForcekN + 20.0f)) * ((MpS.ToKpH(AbsSpeedMpS) + 150.0f) / (2.0f * MpS.ToKpH(AbsSpeedMpS) + 150.0f));
+                    k1 = 0.055f;
+                    k2 = 200f;
+                    k3 = 50f;
+                    k4 = 150f;
+                    k5 = 75f;
+
+                    frictionfraction = k1 * ((brakeShoeForcekN + k2) / (brakeShoeForcekN + k3)) * ((MpS.ToKpH(AbsSpeedMpS) + k4) / (MpS.ToKpH(AbsSpeedMpS) + k5));
+                }
+                else if (BrakeShoeType == BrakeShoeTypes.Disc_Pads)
+                {
+                    k1 = 0.385f;
+                    k2 = -24.5f;
+                    k3 = -27.2f;
+                    k4 = 39.5f;
+                    k5 = 33f;
+
+                    frictionfraction = k1 * ((brakeShoeForcekN + k2) / (brakeShoeForcekN + k3)) * ((MpS.ToKpH(AbsSpeedMpS) + k4) / (MpS.ToKpH(AbsSpeedMpS) + k5));
```

Comment:
You are using the same formula in several places. Consider moving it outside the if blocks to avoid code duplication